src/ws/frame.rs

```diff
@@ -29,24 +28,22 @@ impl Frame {
 
     /// Create a new Close control frame.
     #[inline]
-    pub fn close(code: CloseCode, reason: &str, genmask: bool) -> Binary {
-        let raw: [u8; 2] = unsafe {
-            let u: u16 = code.into();
-            mem::transmute(u.to_be())
-        };
-
-        let payload = if let CloseCode::Empty = code {
-            Vec::new()
-        } else {
-            Vec::from_iter(
-                raw[..]
-                    .iter()
-                    .chain(reason.as_bytes().iter())
-                    .cloned(),
-            )
-        };
-
-        Frame::message(payload, OpCode::Close, true, genmask)
+    pub fn close(reason: Option<CloseReason>, genmask: bool) -> Binary {
+	    let payload = match reason {
```

Comment:
Some indentations seem odd, mixing space and tab I guess? It would be better to use the same style as the rest of the codebase.

Reply:
`cargo fmt` should help

Reply:
`cargo fmt` tried to change a lot of code, including stuff I didn't touch, so I just manually fixed this.

Reply:
Do you use rustfmt-preview?

Reply:
I tried rustfmt-preview, and also the rustfmt built into the IntelliJ Rust plugin (Not sure how it's different) but they both tried to change a lot of code I didn't touch.